internal/core/data/application/eventpurge.go

```diff
@@ -0,0 +1,308 @@
+//
+// Copyright (C) 2025 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package application
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/data/container"
+	pgModels "github.com/edgexfoundry/edgex-go/internal/pkg/infrastructure/postgres/models"
+
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v4/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v4/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/v4/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v4/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v4/errors"
+)
+
+var asyncPurgeEventOnce sync.Once
```

Comment:
since CoreDataApp is used by all the application layer, move the global variable to the CoreDataApp struct

Reply:
Done